grades/tasks_pytest_test.py

```diff
@@ -60,27 +73,29 @@ def test_generate_course_certificates():
     )
     ProctoredExamGradeFactory.create_batch(
         2,
-        user=factory.Iterator([final_grade.user for final_grade in final_grades_with_passed_exam_no_grades]),
+        user=factory.Iterator([final_grade.user for final_grade in passed_final_grades_with_exam[2:4]]),
         course=course_with_exams,
         exam_run=exam_run_no_grades,
         passed=True,
     )
     ProctoredExamGradeFactory.create_batch(
         2,
-        user=factory.Iterator([final_grade.user for final_grade in final_grades_with_failed_exam]),
+        user=factory.Iterator([final_grade.user for final_grade in passed_final_grades_with_exam[4:6]]),
         course=course_with_exams,
         passed=False,
     )
-
+    # course runs need to have CourseRunGradingStatus to get certificates
+    for final_grade in all_final_grades:
```

Comment:
This seems to be the only usage of `all_final_grades`. I think it's probably easier and more readable to get rid of that list and instead just query `FinalGrade` here and fetch the associated `course_run`s.